server/utils/util.go

```diff
@@ -9,20 +10,25 @@ import (
 )
 
 // HashPassword hashes user password
-func HashPassword(user *models.User) {
+func HashPassword(user *models.User) error {
+	hashError := errors.New("unable to hash user password")
 	bytes, err := bcrypt.GenerateFromPassword([]byte(user.Password), 10)
 	if err != nil {
 		fmt.Printf("Unable to hash password " + err.Error())
+		return hashError
 	}
 	user.Password = string(bytes)
+	return nil
 }
 
-func CreateHashedPassword(password string) string {
+func CreateHashedPassword(password string) (string, error) {
+	createError := errors.New("unable to create hashed password")
```

Comment:
this function is supposed to only take in password string as input so we will have no access to which particular user's password it is helping to hash, so is it ok if we don't include username for this error?

Reply:
It doesn't make sense to print this error but we don't know whose account is it. Parse in the user name as a parameter and include the user name for the error.

Reply:
If that's the case, would this function be redundant? Since I already have the function `HashPassword` above that takes in a `models.User` struct which has both the NUSNETID and the password.

Reply:
Yes you can just use the HashPassword instead.
Change the CheckPasswordHash parameter to take in models.User struct